aidial_assistant/commands/run_tool.py

```diff
@@ -81,17 +41,26 @@ async def execute(
     ) -> ResultObject:
         query = get_required_field(args, "query")
 
-        ops = collect_operations(
-            self.plugin.info.open_api, self.plugin.info.ai_plugin.api.url
-        )
+        spec = self.plugin.info.open_api
+        spec_url = self.plugin.info.get_full_spec_url()
 
-        def create_command_tool(op: APIOperation) -> CommandTool:
-            return lambda: OpenAPIChatCommand(
-                op, self.plugin.auth
-            ), _construct_tool(op)
+        def create_command(operation: APIOperation) -> CommandConstructor:
+            # The function is necessary to capture the current value of op.
+            # Otherwise, only first op will be used for all commands
+            return lambda: OpenAPIChatCommand.create(
+                spec_url, operation, self.plugin.auth
+            )
 
         commands: CommandToolDict = {
-            name: create_command_tool(op) for name, op in ops.items()
+            operation.operation_id: (create_command(operation), tool)
```

Comment:
Hard to read: three for's where only two are required.

Maybe rewrite it as two explicit for-loops.